src/backend/opencl/program.cpp

```diff
@@ -67,4 +69,32 @@ namespace opencl
             throw;
         }
     }
+
+    NDRange getKernelLaunchCfg1D(const Kernel& kernel)
+    {
+        int device = getActiveDeviceId();
+
+        size_t dim0ThreadCount = getDeviceWorkGroupSize(kernel, device);
+
+        // make sure you are not launching what the device can't handle
+        dim0ThreadCount = std::min(getDevice(device).getInfo<CL_DEVICE_MAX_WORK_GROUP_SIZE>(), dim0ThreadCount);
```

Comment:
I can't give you example case right away, but i added it as a cautionary check to warrant unexpected values.